File: lista/Lista Dinâmica/Java/Lista/Listavel.java
public interface Listavel {
	
	//anexa(insere ao final da lista) a lista o novo dado fornecido
	void anexar(Object dado) throws OverflowException;
	
	//insere o novo dado fornecido na lista, numa posicao logica informada 
	void inserir(int posicao, Object dado) throws OverflowException;
	
	//retorna o dado que está numa posicao logica informada
	Object selecionar(int posicao);
	
	//retorna todos os dado da ED
	Object[] selecionarTodos();	
	//substitui o dado de uma posicao logica informada,
	//pelo novo dado fornecido
	void atualizar(int posicao, Object novodado);
	
	//remove o dado de uma posicao logica informada
	Object apagar(int posicao) throws UnderflowException;
	
	//métodos auxiliares
	boolean estaCheia();
	boolean estaVazia();
	String imprimir();
}

File: lista/dinamica/Kotlin/Lista/Listavel.kt
interface Listavel {
    // Metodos principais
    fun anexar(dado: Any?)                  //C
    fun inserir(posicao: Int, dado: Any?)   //C
    fun selecionar(posicao: Int): Any?      //R
    fun selecionarTodos(): Array<Any?>      //R
    fun atualizar(posicao: Int, dado: Any?) //U
    fun apagar(posicao: Int): Any?          //D
    fun apagarTodos(): Array<Any?>          //D
    fun limpar()                            //D
    
	//fun juntar(lista1: Listavel, lista2: Listavel): Listavel
	//fun subLista(inicio: Int, fim: Int): Listavel
    // Metodos auxiliares
    fun tamanho(): Int  
    fun estaCheia(): Boolean
    fun estaVazia(): Boolean
    fun imprimir(): String
}